utilities/xmlrpcpp/src/XmlRpcClient.cpp

```diff
@@ -458,15 +459,25 @@ XmlRpcClient::parseResponse(XmlRpcValue& result)
     if ( ! result.fromXml(_response, &offset)) {
       XmlRpcUtil::error("Error in XmlRpcClient::parseResponse: Invalid response value. Response:\n%s", _response.c_str());
       _response = "";
+      _header = "";
       return false;
     }
   } else {
     XmlRpcUtil::error("Error in XmlRpcClient::parseResponse: Invalid response - no param or fault tag. Response:\n%s", _response.c_str());
     _response = "";
+    _header = "";
     return false;
   }
       
   _response = "";
+
+  // Close connection if protocol is HTTP/1.0
+  if (_header.rfind("HTTP/1.0", 0) == 0) {
```

Comment:
Why use `rfind` here? Existing code uses this: https://github.com/ros/ros_comm/blob/c6ad71e7105f151d912c8c0db3bd18caad771e89/utilities/xmlrpcpp/src/XmlRpcServerConnection.cpp#L135

Reply:
I was checking the code and actually _header contains much more than you'd expect:
```
HTTP/1.1 200 OK
Server: BaseHTTP/0.3 Python/2.7.17
Date: Mon, 10 Aug 2020 19:48:04 GMT
Content-type: text/xml
Content-length: 274

<?xml version='1.0'?>
<methodResponse>
<params>
<param>
<value><array><data>
<value><int>1</int></value>
<value><string>Unregistered [/talker] as provider of [/rosout]</string></value>
<value><int>1</int></value>
</data></array></value>
</param>
</params>
</methodResponse>
```

Is what _header contains when I run this branch locally on my melodic machine. A plain find would potentially find HTTP/1.0 randomly inside the xml payload. I've no clue why _header contains much more than just the header…

I actually don't have a kinetic machine at hand. Our robots are also on melodic. Is there any chance somebody else could take over here?

Reply:
I see now. So the `rfind` is used to mimic "startswith" and avoid searching through the rest of the string. Sounds good to me. (Can't wait for `starts_with` in C++20...)